source/module_io/write_wfc_r.cpp

```diff
@@ -31,157 +30,173 @@ void write_psi_r_1(const psi::Psi<std::complex<double>>& wfc_g,
 #endif
-		for(int ik=0; ik<wfc_g.get_nk(); ++ik)
-		{
-			wfc_g.fix_k(ik);
-			const int ik_out = (PARAM.inp.nspin!=2)
-				? ik + GlobalC::Pkpoints.startk_pool[GlobalV::MY_POOL]
-				: ik - kv.get_nks()/2*kv.isk[ik] + kv.get_nkstot()/2*kv.isk[ik] + GlobalC::Pkpoints.startk_pool[GlobalV::MY_POOL];
-			for(int ib=0; ib<wfc_g.get_nbands(); ++ib)
-			{
-				const std::vector<std::complex<double>> wfc_r = cal_wfc_r(wfcpw, wfc_g, ik, ib);
-
-                std::vector<double> wfc_r2(wfc_r.size());
-                std::vector<double> wfc_i2;
-                if (square)
-                    for (int ir = 0; ir < wfc_r2.size(); ++ir)
-                        wfc_r2[ir] = std::norm(wfc_r[ir]);   // "std::norm(z)" returns |z|^2
-                else
+    for (int ik = 0; ik < wfc_g.get_nk(); ++ik)
+    {
+        wfc_g.fix_k(ik);
+        const int ik_out = (PARAM.inp.nspin != 2)
+                               ? ik + GlobalC::Pkpoints.startk_pool[GlobalV::MY_POOL]
+                               : ik - kv.get_nks() / 2 * kv.isk[ik] + kv.get_nkstot() / 2 * kv.isk[ik]
+                                     + GlobalC::Pkpoints.startk_pool[GlobalV::MY_POOL];
+        for (int ib = 0; ib < wfc_g.get_nbands(); ++ib)
+        {
+            const std::vector<std::complex<double>> wfc_r = cal_wfc_r(wfcpw, wfc_g, ik, ib);
+
+            std::vector<double> wfc_r2(wfc_r.size());
+            std::vector<double> wfc_i2;
+            if (square)
+            {
+                for (int ir = 0; ir < wfc_r2.size(); ++ir)
                 {
-                    wfc_i2.resize(wfc_r.size());
-                    for (int ir = 0; ir < wfc_r2.size(); ++ir)
-                    {
-                        wfc_r2[ir] = wfc_r[ir].real();
-                        wfc_i2[ir] = wfc_r[ir].imag();
-                    }
+                    wfc_r2[ir] = std::norm(wfc_r[ir]); // "std::norm(z)" returns |z|^2
                 }
-				const std::string file_name = outdir + "wfc_realspace_"
-					+ ModuleBase::GlobalFunc::TO_STRING(ik_out)
-					+ "_" + ModuleBase::GlobalFunc::TO_STRING(ib);
+            }
+            else
+            {
+                wfc_i2.resize(wfc_r.size());
+                for (int ir = 0; ir < wfc_r2.size(); ++ir)
+                {
+                    wfc_r2[ir] = wfc_r[ir].real();
+                    wfc_i2[ir] = wfc_r[ir].imag();
+                }
+            }
+            const std::string file_name = outdir + "wfc_realspace_" + ModuleBase::GlobalFunc::TO_STRING(ik_out) + "_"
+                                          + ModuleBase::GlobalFunc::TO_STRING(ib);
 #ifdef __MPI
-				mpi_requests.push_back({});
-                write_chg_r_1(wfcpw, wfc_r2, file_name, mpi_requests.back());
-                if (!square)
-                    write_chg_r_1(wfcpw, wfc_i2, file_name + "_imag", mpi_requests.back());
+            mpi_requests.push_back({});
+            write_chg_r_1(wfcpw, wfc_r2, file_name, mpi_requests.back());
```

Comment:
in function `write_psi_r_1` you call `write_chg_r_1`, this action is confusing and I know it is because the name of `write_chg_r_1` is not a good idea, could you have some improvements here?

Reply:
I added an annotation here to explain the use of `write_chg_r_1`.